src/services/service.js

```diff
@@ -25,6 +25,20 @@ class Service {
     return query;
   }
 
+  async countAllBySearch(search) {
+    let query, count;
+
+    if (search) {
+      query = await this.Model.searchByKeyword(search);
```

Comment:
so you want to get the count right, then why not just find the count, instead of find all the documents then get the count. Have a look of the docs for mongodb and mongoose